sql/src/main/java/io/crate/expression/predicate/PredicateModule.java

```diff
@@ -21,34 +21,13 @@
 
 package io.crate.expression.predicate;
 
-import io.crate.metadata.FunctionIdent;
+import io.crate.expression.AbstractFunctionModule;
 import io.crate.metadata.FunctionImplementation;
-import io.crate.metadata.FunctionName;
-import io.crate.metadata.FunctionResolver;
-import org.elasticsearch.common.inject.AbstractModule;
-import org.elasticsearch.common.inject.multibindings.MapBinder;
 
-public class PredicateModule extends AbstractModule {
-
-    private MapBinder<FunctionIdent, FunctionImplementation> functionBinder;
-    private MapBinder<FunctionName, FunctionResolver> resolverBinder;
-
-    public void register(FunctionImplementation impl) {
-        functionBinder.addBinding(impl.info().ident()).toInstance(impl);
-    }
-
-    public void register(String name, FunctionResolver functionResolver) {
-        register(new FunctionName(name), functionResolver);
-    }
-
-    public void register(FunctionName qualifiedName, FunctionResolver functionResolver) {
-        resolverBinder.addBinding(qualifiedName).toInstance(functionResolver);
-    }
+public class PredicateModule extends AbstractFunctionModule<FunctionImplementation> {
```

Comment:
I extended the `PredicateModule` from `AbstractFunctionModule` or there was a reason maybe why it wasn't done like that before?

Reply:
I don't see any reasons why this was not extended. Maybe just never refactored...